code/modules/projectiles/guns/energy/nuclear.dm

```diff
@@ -66,24 +66,24 @@
 				M << "<span class='danger'>Your gun's reactor overloads!</span>"
 			M << "<span class='warning'>You feel a wave of heat wash over you.</span>"
 			M.apply_effect(300, IRRADIATE)
-		crit_fail = 1 //break the gun so it stops recharging
+		//crit_fail = 1 //break the gun so it stops recharging
 		processing_objects.Remove(src)
 		update_icon()
 	return 0
 
 
 /obj/item/weapon/gun/energy/gun/nuclear/proc/update_charge()
-	if (crit_fail)
-		overlays += "nucgun-whee"
-		return
+	// if (crit_fail)

```

Comment:
The nuke gun actually had functionality that wasn't refactored. So declaring `/obj/item/weapon/gun/energy/gun/nuclear/var/crit_fail = 0` or some other var name, and using that, would be great. Similarly to how it was done with cells and other uses of the more general crit_fail.